### PR TITLE
[MERGE_MINPY] add log.py, a more verbose customed logger

### PR DESCRIPTION
**Pull request by ZihengJiang**
Migrated from MinPy

Usage: 
```
from mxnet.log import mxlogger
# or you can use it to replace the logging directly like:
# from mxnet.log import mxlogger as logging

mxlogger.info("some message")
```

I set a global logger in `mxnet.log` module called `mxlogger`, and this logger will not change the global python logging module state.

I'm not sure whether to replace all the `logging` used by the other file. Because in some situation, they're just used to print some message, like https://github.com/dmlc/mxnet/blob/master/python/mxnet/model.py#L346. But when you want to debug something in python, a more verbose logger will be useful.

@piiswrong

### CODE REVIEW

File: python/mxnet/log.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    """
+    logger = logging.getLogger(name)
+    if name and not getattr(logger, '_init_done', None):
+        logger._init_done = True
+        logger.addHandler(_handler)
+        logger.setLevel(level)
+    return logger
+
+
+mxlogger = _get_logger(name='mxnet', level=WARNING)
```
Comment (piiswrong) [inline python/mxnet/log.py]: Config default logging here. Also add a function to reconfig and add a output logging file

**Comment by ZihengJiang** [comment]
use the root logger as default (not sure it is a good idea) and add `basicConfig` in order to forward config

**Comment by piiswrong** [comment]
Could you remove 
```
+_handler = logging.StreamHandler()
 +_handler.setFormatter(_Formatter())
 +
 +# get the root logger
 +mxlogger = getLogger(name='', level=WARNING)
 +
 +def basicConfig(**kwarg):
 +    """Forward to the logging.basicConfig"""
 +    mxlogger.handlers = []
 +    # This function does nothing if the root logger
 +    # already has handlers configured.
 +    logging.basicConfig(**kwarg)
```
So it doesn't have any effect by default. I'll verify it doesn't confuse users later

**Comment by ZihengJiang** [comment]
Agree with the removing.

**Pull request merged**